contracts-bedrock: cancun evm version

tynes opened the pull request:
**Description**

Turns on the cancun evm version for the evm now that
decun is on mainnet. Also ignores some compiler warnings
related to tstore, code too large and initcode too large.

<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Added support for specifying EVM version "Cancun" and ignoring specific error codes in contract configurations.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tynes committed:
contracts-bedrock: cancun evm version

Turns on the cancun evm version for the evm now that
decun is on mainnet. Also ignores some compiler warnings
related to tstore, code too large and initcode too large.

mds1 commented on packages/contracts-bedrock/foundry.toml:
```diff
@@ -25,6 +25,8 @@ remappings = [
 bytecode_hash = 'none'
 build_info_path = 'artifacts/build-info'
+evm_version = "cancun"
+ignored_error_codes = [2394, 5574, 3860]
```
This should work and is more readable ([source](https://github.com/foundry-rs/foundry/blob/bf5bfe0a3af293152f84f965e21856bc740aac0d/crates/config/src/error.rs#L152))
```suggestion
ignored_error_codes = ["transient-storage", "code-size", "init-code-size"]
```

tynes committed:
fix: human readable config

Updates the `ignored_error_codes` section in foundry.toml to be more human readable

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

tynes commented:
Build is failing with:

```
foundry config error: Unknown variant transient-storage for setting `ignored_error_codes.0`
```

Maybe its an older version of foundry we are on

clabby reviewed (approved):
hype

Inphi commented:
I updated the PR to pull in the pnpm version upgrade.

Inphi commented:
This failing because the `ci` build profile treats warnings as errors. But forge doesn't highlight any warnings when run locally without `deny_warnings` set: 
```
$ env FOUNDRY_PROFILE=ci forge build
[⠒] Compiling...
[⠑] Compiling 16 files with 0.8.19
[⠘] Compiling 124 files with 0.8.25
[⠃] Compiling 357 files with 0.8.15
[⠒] Compiling 1 files with 0.5.17
[⠘] Solc 0.5.17 finished in 58.70ms
[⠢] Solc 0.8.19 finished in 658.01ms
[⠔] Solc 0.8.25 finished in 5.16s
[⠘] Solc 0.8.15 finished in 179.06s
Compiler run successful with warnings:
$ 
```
Seems to be a foundry bug.
